### ISSUE TITLE
ConsoleReader.readLine(null, ConsoleReader.NULL_MASK) results in a blank line.

### ISSUE THREAD

**Issue by nrg4878** [jline/jline2#181]
ConsoleReader.readLine(null, ConsoleReader.NULL_MASK) results in a blank line.
The above API call  prevents the line to be echoed back to the terminal. However on the last character of the line, the accept() calls println() which blindly prints a newline character.
    public final void println() throws IOException {
          print(CR);
    }

Something like this will print the newline char IFF the mask is null or some other character specified by the caller. The newline char is not needed when NULL_MASK is used.

    public final void println() throws IOException {
        if (mask == null)
          print(CR);
    }

**Comment by nrg4878** [comment]
To clarify, readLine() returns the line read from the file to the caller. However, it prints a blank line to the terminal before returning to caller.

**Comment by gnodet** [comment]
I think I understand you're problem, but isn't that also conditioned to the fact that there's no prompt ?
Also I think println() is used in various cases, so it may be better to do the test in accept() / abort() instead of println() directly, as it's done in putString(), insert() or moveInternal()...

**Comment by nrg4878** [comment]
Hey Guillaume,
Thank you for attending to this issue. jline is used in SQLLine/Beeline (Apache Hive) that uses it to read multi-line SQL queries (sometimes spread over dozens of lines in the file). Using beeline in non-interactive mode is popular mode where it reads the query to a file and the query result is written out to a file (either via stdout redirect or a commandline option). Having a non-deterministic number of blankl lines in this output 
a) Effects readability of the results.
b) Makes it hard to parse the results.

I agree with you about having this fix made to println() if it is used elsewhere. I leave it to your expertise to find the best fix. Thanks again

**Comment by nrg4878** [comment]
Hey Guillaume,
Any update on this issue? Any idea when you might be able to get to this? Thanks

**Comment by gnodet** [comment]
Can you provide a patch to disable the println() iff there's no prompt and a null mask is used ?

**Comment by Ocean22** [comment]
Hi,my friends,is there any progress on this issue? `beeline` client will get many blank rows,need to fix it.@gnodet @nrg4878 https://github.com/apache/kyuubi/issues/5753
